### ISSUE TITLE
Inconsistent/Broken JPX support

### ISSUE THREAD

**Issue by adam-vessey** [cantaloupe-project/cantaloupe#320]
We have come across some JPX files masquerading as JP2s in the wild; in particular, it appears that Adobe Photoshop's JPEG 2000 plugin:

> [...] creates JPEG 2000 Part 2 files (JPX) with Part 1 (JP2) compatibility

Looking into [discerning things](http://wiki.opf-labs.org/display/TR/Files+identified+as+JP2+are+really+JPX), and the are indeed not _strictly_ JP2s according to Jpylyzer; however, OpenJPEG itself seems able to decode them.

At issue is [more recent expanded signature check](https://github.com/cantaloupe-project/cantaloupe/blob/98b962ce58693ba1953eff4ee8d0317bbe214d77/src/main/java/edu/illinois/library/cantaloupe/processor/codec/JPEG2000MetadataReader.java#L113-L116) such that these files are now rejected as invalid.

There _does_ seem to be an [intent to support JPX files in the `OpenJpegProcessor`](https://github.com/cantaloupe-project/cantaloupe/blob/98b962ce58693ba1953eff4ee8d0317bbe214d77/src/main/java/edu/illinois/library/cantaloupe/processor/OpenJpegProcessor.java#L385-L386); however, if they had similarly varying signatures, I suspect they would similarly be rejected.

One of these JPXs:
```
00000000  00 00 00 0c 6a 50 20 20  0d 0a 87 0a 00 00 00 1c  |....jP  ........|
00000010  66 74 79 70 6a 70 32 20  00 00 00 00 6a 70 32 20  |ftypjp2 ....jp2 |
00000020  6a 70 78 62 6a 70 78 20  00 00 00 b5 72 72 65 71  |jpxbjpx ....rreq|
```

More typical JP2 (matching the present signature):
```
00000000  00 00 00 0c 6a 50 20 20  0d 0a 87 0a 00 00 00 14  |....jP  ........|
00000010  66 74 79 70 6a 70 32 20  00 00 00 00 6a 70 32 20  |ftypjp2 ....jp2 |
00000020  00 00 00 2d 6a 70 32 68  00 00 00 16 69 68 64 72  |...-jp2h....ihdr|
```

... the issue is there at byte `0x0f`, being `0x1c` vs the `0x14` in the signature we check for.

... I've not looked too deep into the JP2 spec, but... is `0x0c` to `0x0f` supposed to be a pointer of some type? If so, it's pointing at an occurrence of `jp2 ` in each case... [There seems to be some support of this interpretation](http://fileformats.archiveteam.org/wiki/JP2#Identification); however, nothing definitive, but I digress...

## TL:DR: Some JPX files used to work but now do not. Thoughts of how to resolve?

**Comment by adolski** [comment]
I have absolutely no memory of why I tightened the signature check in a3527141f70b8c0fe7196ace64090efc464e258e. That check is too tight according to the JP2 spec so I re-loosened it and the tests still pass... 🤷‍♂ Maybe whatever broke will break again and then I can fix it properly next time.

**Comment by adam-vessey** [comment]
Awesome, thanks!
